### ISSUE TITLE
`WriteDescriptorSet` expects to only handle `image_info`

### ISSUE THREAD

**Issue by Seelengrab** [JuliaGPU/Vulkan.jl#24]
According to vulkan-tutorial.com, `VkWriteDescriptorSet` can either refer to buffers, image data or buffer views. The wrapper `WriteDescriptorSet` defines the field `descriptorCount` of `VkWriteDescriptorSet` solely based on the length of `image_info`, which is required to be passed, even though it may not be used every time the wrapper is called.

I could trace this down to this line in vulkan_wrapper.jl containing `pointer_length(image_info)`:

https://github.com/JuliaGPU/Vulkan.jl/blob/1190a02b309635f12ba08021cdd0ef6a3b3a6b05/generated/vulkan_wrapper.jl#L10888

which imo should be `max(pointer_length(image_info), pointer_length(buffer_info), pointer_length(texel_buffer_view))`.

Another option would be to make these three arguments keywords and check which exists (checking for exclusivity as well) and using that as the basis for the length.

**Comment by Seelengrab** [comment]
Ref. [vulkan-tutorial.com](https://vulkan-tutorial.com/Uniform_buffers/Descriptor_pool_and_sets):

 ```C++
descriptorWrite.pBufferInfo = &bufferInfo;
descriptorWrite.pImageInfo = nullptr; // Optional
descriptorWrite.pTexelBufferView = nullptr; // Optional
```
> The last field references an array with `descriptorCount` structs that actually configure the descriptors. It depends on the type of descriptor which one of the three you actually need to use. The `pBufferInfo` field is used for descriptors that refer to buffer data, `pImageInfo` is used for descriptors that refer to image data, and `pTexelBufferView` is used for descriptors that refer to buffer views. Our descriptor is based on buffers, so we're using pBufferInfo.

**Comment by serenity4** [comment]
Nice catch. The issue is that it's not captured in the XML specification, so we'll need a manual tweak.

**Comment by Seelengrab** [comment]
I think the Vulkan spec is a "live" spec, i.e. it's possible to suggest changes. Can we help out others consuming the XML version by upstreaming this?

**Comment by serenity4** [comment]
We can open an issue in [Vulkan-Docs](https://github.com/KhronosGroup/Vulkan-Docs). The thing is, the XML captures most but not all constraints/specifications, and just declares `noautovalidity=true` for exceptions, like these members of `VkWriteDescriptorSet`. I'll take a look and see to which extent these exceptions span the XML and with what consequences for us, but I don't think we'll escape from needing manual tweaks here and there at some point.

**Comment by serenity4** [comment]
As [per the spec](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VkWriteDescriptorSet), this `descriptorCount` can represent many things, so it's best to just expose it.

**Comment by serenity4** [comment]
On second thoughts, we may add `descriptorCount` as a keyword argument whose default value is the maximum of all three array lengths, as you proposed, so most users won't have to deal with specifying it.

**Comment by serenity4** [comment]
Implemented with 8dfb51b06f387bfc2573d95498587c3339284b54.
